src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,575 @@
+     *     the specified algorithm
+     * @throws NullPointerException
+     *     if the algorithm is {@code null}
+     */
+    public static KDF getInstance(String algorithm)
+        throws NoSuchAlgorithmException {
+        try {
+            return getInstance(algorithm, (AlgorithmParameterSpec) null);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidParameterSpecException. Does this "
+                + "algorithm require an "
+                + "AlgorithmParameterSpec?");
+        }
+    }
+
+    /**
+     * Creates an instance of the {@code KDF} object with a specific
+     * {@code Provider}.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param provider
+     *     the provider to use for this key derivation
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if no {@code Provider} supports a {@code KDFSpi} implementation for
+     *     the specified algorithm
+     * @throws NoSuchProviderException
+     *     if the specified provider is not registered in the security provider
+     *     list
+     * @throws NullPointerException
+     *     if the algorithm is {@code null}
+     */
+    public static KDF getInstance(String algorithm, String provider)
+        throws NoSuchAlgorithmException, NoSuchProviderException {
+        try {
+            return getInstance(algorithm, null,
+                               provider);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidParameterSpecException. Does this "
+                + "algorithm require an "
+                + "AlgorithmParameterSpec?");
+        }
+    }
+
+    /**
+     * Creates an instance of the {@code KDF} object using a supplied
+     * {@code Provider} instance.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param provider
+     *     the provider
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if no {@code Provider} supports a {@code KDFSpi} implementation for
+     *     the specified algorithm
+     * @throws NullPointerException
+     *     if the algorithm is {@code null}
+     */
+    public static KDF getInstance(String algorithm, Provider provider)
+        throws NoSuchAlgorithmException {
+        try {
+            return getInstance(algorithm, null,
+                               provider);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidParameterSpecException. Does this "
+                + "algorithm require an "
+                + "AlgorithmParameterSpec?");
+        }
+    }
+
+    /**
+     * Creates an instance of the {@code KDF} object.
```

Comment:
Suggest rewording as "Returns a {@code KDF} object that implements the specified algorithm and is initialized with the specified parameters."